### PR TITLE
dev/reference: update CHARSET and COLLATION infos

### PR DESCRIPTION
**Pull request by bb7133**
updated docs about `CHARACTER SET` and `COLLATION` in TiDB, this is for the release of 3.0

### CODE REVIEW

**Comment by tangenta** [comment]
LGTM

File: dev/reference/mysql-compatibility.md [hunk 1]
```diff
@@ -126,12 +130,13 @@ TiDB 支持 MySQL 5.7 中 **绝大多数的 SQL 模式**，以下几种模式除
...
-    + MySQL 5.7 中为 `latin1`，但在 MySQL 8.0 中修改为 `utf8mb4`
++ 默认字符集与 MySQL 不同：
+    + TiDB 中为 `utf8mb4`
+    + MySQL 5.7 中为 `latin1`，MySQL 8.0 中修改为 `utf8mb4`
 + 默认排序规则不同：
-    + MySQL 5.7 中使用 `latin1_swedish_ci`
-    + TiDB 使用 `binary`
+    + TiDB 中，`utf8mb4` 的默认排序规则为 `utf8mb4_bin`
+    + MySQL 5.7 中，`utf8mb4` 的默认排序规则为 `utf8mb4_general_ci`，MySQL 8.0 中修改为 `utf8mb4_0900_ai_ci`
+    + 请使用 [`SHOW CHARACTER SET`](/dev/reference/sql/statements/admin.md## `SHOW` 语句) 语句查看所有字符集的默认排序规则
```
Comment (lilin90) [inline dev/reference/mysql-compatibility.md]: The link is incorrect here. @yikeke Please help to fix this in another PR. @bb7133 If you're not sure about the format, feel free to involve the I18N members to review it.

File: dev/reference/sql/character-set.md [hunk 2]
```diff
@@ -91,15 +87,19 @@ mysql> SHOW COLLATION WHERE Charset = 'latin1';
...
     [[DEFAULT] CHARACTER SET charset_name]
     [[DEFAULT] COLLATE collation_name]
+
+ALTER DATABASE db_name
+    [[DEFAULT] CHARACTER SET charset_name]
+    [[DEFAULT] COLLATE collation_name]
 ```
 
-在这里 `SCHEMA` 可以跟 `DATABASE` 互换使用。
+在这里 `DATABASE` 可以跟 `SCEHMA` 互换使用。
```
Comment (lilin90) [inline dev/reference/sql/character-set.md]: Typo: SCEHMA -> SCHEMA @bb7133

File: dev/reference/mysql-compatibility.md (hunk 1 above)
Comment (yikeke) [reply dev/reference/mysql-compatibility.md]: Okay.

**Comment by lilin90** [comment]
@yikeke Please help check the content and format in this PR and fix them in a new PR. Thanks!

File: dev/reference/mysql-compatibility.md [hunk 3]
```diff
@@ -80,17 +80,21 @@ TiDB 支持常用的 MySQL 内建函数，但是不是所有的函数都已经
...
-  - 不支持同时创建多个列 
+  - 不支持同时创建多个列
   - 不支持将新创建的列设为主键或唯一索引，也不支持将此列设成 auto_increment 属性
 + Drop Column: 不支持删除主键列或索引列
 + Change/Modify Column
   - 不支持有损变更，比如从 `BIGINT` 变为 `INTEGER`，或者从 `VARCHAR(255)` 变为 `VARCHAR(10)`
   - 不支持更改 `UNSIGNED` 属性
   - 只支持将 `CHARACTER SET` 属性从 `utf8` 更改为 `utf8mb4`
++ Alter Table
+  - 只支持将 `CHARACTER SET` 属性从 `utf8` 更改为 `utf8mb4`
```
Comment (zimulala) [inline dev/reference/mysql-compatibility.md]: 这个跟 line 93 重复了。

Comment (zimulala) [reply dev/reference/mysql-compatibility.md]: 合了，我在我的 PR 里面改一下吧

**Pull request merged**